### PR TITLE
Pin code tasks

### PR DESCRIPTION
**Pull request by Arens404**

### CODE REVIEW

File: src/pinCodeTasks/pinCodeTask2.java [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+
+import java.util.Scanner;
+
+public class pinCodeTask2 {
+    public static void main(String[] args) {
+        try{
+            Scanner userInputScanner = new Scanner(System.in);
+            int correctSafeCode = 777777;
+            int safeCodeCount = 0;
+            while(safeCodeCount != 77777){
```
Comment (Chezo) [inline src/pinCodeTasks/pinCodeTask2.java]: magic number in while

File: src/pinCodeTasks/pinCodeTask2.java [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+                System.out.print("Enter the safe code: ");
+                if(userInputScanner.hasNext("q")){
+                    System.exit(0);
+                } else { }
+                int safeGuessInput = userInputScanner.nextInt();
```
Comment (Chezo) [inline src/pinCodeTasks/pinCodeTask2.java]: why is it safe? this variable name not too good

Comment (Chezo) [reply src/pinCodeTasks/pinCodeTask2.java]: also its not an input, its an integer.

File: src/pinCodeTasks/pinCodeTask1.java [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+                    break;
+                }
+                else if(pinCodeGuessInput <= 999 || pinCodeGuessInput >= 9999){
+                    System.out.println("The pin code is four number!");
+                    break;
+                }
+                else {
+                    System.out.println("Wrong pin!");}
+            }
+        }catch (IllegalArgumentException NumberFormatException){
```
Comment (Chezo) [inline src/pinCodeTasks/pinCodeTask1.java]: so which exception? illegalargument or numberformat? its mislead

**Review by Chezo — CHANGES_REQUESTED** [review]

File: pinCodeMainTasks.java [hunk 4]
```diff
@@ -0,0 +1,78 @@
+import java.util.InputMismatchException;
+import java.util.Scanner;
+
+public class pinCodeMainTasks {
+    public static void main(String[] args) {
+        Scanner scanner = new Scanner(System.in);
```
Comment (Chezo) [inline pinCodeMainTasks.java]: could be final and must be private

File: pinCodeMainTasks.java [hunk 5]
```diff
@@ -0,0 +1,78 @@
...
+        System.out.print("Select witch task u wanna try:");
+        try{
+            int UserInputContainer = scanner.nextInt();
```
Comment (Chezo) [inline pinCodeMainTasks.java]: still int

File: pinCodeMainTasks.java [hunk 6]
```diff
@@ -0,0 +1,78 @@
...
+                case 2 -> pinCodeTaskTwo();
+                default -> System.out.println("U have only two option! Please select from taskOne or taskTwo");
+            }
+        } catch (InputMismatchException IllegalArgumentException){
+            System.out.println("Please use numbers to select a task!");
+        }
+    }
+    public static void pinCodeTaskOne(){
+        try{
+            Scanner userInputScanner = new Scanner(System.in);
```
Comment (Chezo) [inline pinCodeMainTasks.java]: why u have multiple scanner? use function parameter instead!

File: pinCodeMainTasks.java [hunk 7]
```diff
@@ -0,0 +1,78 @@
...
+            int correctPinCode = 5619;
+            int pinCodeGuessCount = 2;
+            int indicatorPinGuessCount = 3;
+            while(pinCodeGuessCount != 5){
```
Comment (Chezo) [inline pinCodeMainTasks.java]: still magic number

File: pinCodeMainTasks.java [hunk 8]
```diff
@@ -0,0 +1,78 @@
...
+                System.out.print("U only have " + indicatorPinGuessCount + " chance\n");
+                System.out.print("Enter your pin code:");
+                int pinCodeGuessInput = Integer.parseInt(userInputScanner.nextLine());
+                pinCodeGuessCount++;
+                indicatorPinGuessCount--;
+                if(pinCodeGuessCount == 5){
```
Comment (Chezo) [inline pinCodeMainTasks.java]: magic number

File: pinCodeMainTasks.java [hunk 9]
```diff
@@ -0,0 +1,78 @@
...
+                    System.out.println("Phone locked!");
+                }
+                else if(pinCodeGuessCount < 5 && pinCodeGuessInput == correctPinCode){
```
Comment (Chezo) [inline pinCodeMainTasks.java]: magic number

File: pinCodeMainTasks.java [hunk 10]
```diff
@@ -0,0 +1,78 @@
...
+                    System.out.println("Phone unlocked!");
+                    break;
+                }
+                else if(pinCodeGuessInput <= 999 || pinCodeGuessInput >= 9999){
```
Comment (Chezo) [inline pinCodeMainTasks.java]: magic numbers

File: pinCodeMainTasks.java [hunk 11]
```diff
@@ -0,0 +1,78 @@
...
+                else {
+                    System.out.println("Wrong pin!");}
+            }
+        }catch (IllegalArgumentException NumberFormatException){
+            System.out.println("Wrong format! Please use numbers to enter the input!");
+        }
+    }
+    public static void pinCodeTaskTwo(){
+        try{
+            Scanner userInputScanner = new Scanner(System.in);
```
Comment (Chezo) [inline pinCodeMainTasks.java]: one more scanner?

File: pinCodeMainTasks.java [hunk 12]
```diff
@@ -0,0 +1,78 @@
...
+                if(userInputScanner.hasNext("q")){
+                    System.exit(0);
+                } else { }
+                UserGuessInput = userInputScanner.nextInt();
+                {
+                    if(UserGuessInput == correctSafeCode){
+                        System.out.println("Safe unlocked!");
+                        System.exit(0);
+                    }
+                    else if(UserGuessInput <= 99999 || UserGuessInput >= 999999){
```
Comment (Chezo) [inline pinCodeMainTasks.java]: magic numbers

File: pinCodeMainTasks.java [hunk 13]
```diff
@@ -0,0 +1,78 @@
...
+                        System.out.println("The safe code is six number!");
+                        System.exit(0);
+                    }
+                    else {
+                        System.out.println("Wrong Code!");
+                    }
+                }
+            }
+        } catch (NumberFormatException IllegalArgumentException){
```
Comment (Chezo) [inline pinCodeMainTasks.java]: mislead name-type pair

**Review by Chezo — CHANGES_REQUESTED** [review]

File: pinCodeMainTasks.java [hunk 14]
```diff
@@ -0,0 +1,79 @@
+import pinCodeTasks.Setter;
+
+import java.util.Scanner;
+
+public class pinCodeMainTasks {
+    private Setter scanner = new Setter();
```
Comment (Chezo) [inline pinCodeMainTasks.java]: setter or scanner? wtf?

File: pinCodeMainTasks.java [hunk 15]
```diff
@@ -0,0 +1,79 @@
...
+    public static void main(String[] args) {
+        pinCodeMainTasks pinCodeTasks = new pinCodeMainTasks();
+        System.out.print("Select witch task u wanna try:");
+        try{
+            int UserInputSelectionInt = pinCodeTasks.scanner.input();
```
Comment (Chezo) [inline pinCodeMainTasks.java]: wtf is this call?

**Review by Chezo — CHANGES_REQUESTED** [review]

File: pinCodeMainTasks.java (hunk 15 above)
Comment (Arens404) [reply pinCodeMainTasks.java]: 4a16cda1

**Review by Chezo — CHANGES_REQUESTED** [review]
please eliminate the extra class with scanner initialization.

**Comment by Arens404** [comment]
> please eliminate the extra class with scanner initialization.

b32293db

**Review by Chezo — CHANGES_REQUESTED** [review]
2 scanners, eliminate one of them

**Comment by Arens404** [comment]
> 2 scanners, eliminate one of them

c1b764f3

**Review by Chezo — APPROVED** [review]
acceptable

**Pull request merged**